packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -83,6 +83,12 @@ function describeLiteralType(type: ts.Type): string {
   }
 
   if (type.isLiteral()) {
+    if (typeof type.value === 'object') {
+      // Print ts.PseudoBigInt
+      return `${
+        type.value.negative ? '-' : ''
+      }${type.value.base10Value.toString()}n`;
+    }
     return type.value.toString();
```

Comment:
This should be handled instead by the condition on line 111
We should move that branch up instaed

Reply:
[43ae7de](https://github.com/typescript-eslint/typescript-eslint/pull/6570/commits/43ae7de86dd9249ab88fecb5659d35e31d2ac7ed)